src/client/actions/ThreadListSync.js

```diff
@@ -0,0 +1,63 @@
+'use strict';
+
+const Action = require('./Action');
+const Collection = require('../../util/Collection');
+const { Events } = require('../../util/Constants');
+
+class ThreadListSyncAction extends Action {
+  handle(data) {
+    const client = this.client;
+
+    let guild = client.guilds.cache.get(data.guild_id);
+    if (guild) {
+      if (!data.channels_ids) {
+        guild.channels.cache.forEach(channel => {
+          this.removeStale(channel);
+        });
+      } else {
+        data.channel_ids.forEach(id => {
+          const channel = client.channels.resolve(id);
+          if (channel) this.removeStale(channel);
+        });
+      }
+
+      const syncedThreads = new Collection();
+      data.threads.forEach(rawThread => {
+        const thread = client.channels.add(rawThread);
+        syncedThreads.set(thread.id, thread);
+      });
+
+      Object.entries(data.members).forEach(([threadID, rawMember]) => {
+        const thread = client.channels.cache.get(threadID);
+        if (thread) {
+          thread.members.add(rawMember);
+        }
+      });
+
+      /**
+       * Emitted whenever the client user gains access to a text or news channel
```

Comment:
What isn't accurate about this?

Reply:
they're not text or news channels, they're thread channels :P

Reply:
But the event is emitted when you gain access to the parent channel, being a text or news channel, even if there are no threads.

Reply:
> But the event is emitted when you gain access to the parent channel, being a text or news channel, even if there are no threads.

The event is WHAT?

Reply:
> even if there are no threads.

rly? I think every place that sends it is checking if there are threads, how'd you reproduce it!

Reply:
Hmm, cannot reproduce with no threads anymore. Although, it does seem weird to not sync threads when there were previously threads. E.g. if you have active threads, lose access to a channel, thread is marked archived, gain access to the channel. Since there is no `THREAD_DELETE` fired here and we obviously don't get `THREAD_UPDATE` while not having access to the channel, the cache is now out of date and doesn't get updated (in this case, removing the now archived threads